build-automation.js

```diff
@@ -0,0 +1,100 @@
+const util = require("util")
+
+const exec = util.promisify(require("child_process").exec);
+
+// a function that queries the Node.js release website for new versions,
+// compare the available ones with the ones we use in this repo
+// and returns whether we should update or not
+const checkIfThereAreNewVersions = async () => {
+  try {
+    const nodeWebsite = await (await fetch('https://nodejs.org/en/download/releases/')).text();
```

Comment:
can we just get the json from https://nodejs.org/download/release/index.json here? to more closely mirror the unofficial build flow.

Reply:
and avoid the quite horrible regex a but below 🙈

Reply:
Yeah, I guess we can. I just thought that the unofficial builds JSON might be updated just as late as the musl builds were finished. That's why I went for the official site.

Do you know if the unofficial builds JSON is updated right away when Node.js releases a new feature? If so, I can make the change. If not, I think we'll probably still need to get that info from the official website.

As for the horrible regex, I'd like to get feedback on how you would avoid it.

Reply:
@Pehesi97 I'm linking to the official website. Unofficial is later, so you need to check both like you do today. But instead of parsing HTML, we can fetch JSON directly

Reply:
Sure. Thank you.